numpy/ma/core.py

```diff
@@ -1682,10 +1689,29 @@ def make_mask_none(newshape, dtype=None):
 
     """
     if dtype is None:
-        result = np.zeros(newshape, dtype=MaskType)
+        dtype = MaskType
     else:
-        result = np.zeros(newshape, dtype=make_mask_descr(dtype))
-    return result
+        dtype = make_mask_descr(dtype)
+
+    if not readonly:
+        return np.zeros(newshape, dtype=dtype)
+
+    else:
+        # build a buffer of the single scalar mask
+        if dtype == MaskType:
+            buff = np.ma.nomask
+        else:
+            buff = np.zeros((), dtype=dtype)
+            buff.flags.writeable = False
+
+        # duplicate it using zero strides
+        # like np.lib.stride_tricks.as_strided, but faster
+        return np.ndarray(
```

Comment:
You're right, it is. Using `buff = b'\0'` instead of `np.ma.nomask` is also faster.

With both optimizations in place, this breaks even at shape `(2000,)`.

For small n, this line by itself is worse by a factor of 2 (600ns)

For `n = 100000`, this is better by a factor of 8.